markdown/htmlparser.py

```diff
@@ -56,6 +56,10 @@ class HTMLExtractor(htmlparser.HTMLParser):
     def __init__(self, md, *args, **kwargs):
         if 'convert_charrefs' not in kwargs:
             kwargs['convert_charrefs'] = False
+
+        # Block tags that should contain no content (self closing)
+        self.empty_tags = ['hr']
```

Comment:
You seem to think we need to treat `<hr>` specially. However, I would expect all 'block-level tags' to behave the same way. Therefore `<script />` would be no different. We only need to check if a tag is a block-level tag in `handle_empty_tags`. No need to specifically list empty tags separately.

Reply:
`<hr>` is a different case than `<script />`. The latter is giving a hint that it is self closed.

`<hr>` gives no hint, and usually, you would keep looking for an ending tag. But `<hr>` *never* has content. Maybe I'm missing something, but this is a very special case.

Reply:
In fact, given:

```
<hr>
some text
</hr>
```

You should treat them both as separate `<hr>` like browsers do.

Reply:
I'm operating no the assumption that the HtmlParser properly recognizes that `<hr>` is a self-closing tag. If that is not the case, then that might be different.

Reply:
Sure enough, the HtmlParser does not recognize `<hr>` as a self-closing tag.

```python
>>> from test.test_htmlparser import EventCollector
>>> ec = EventCollector()
>>> ec.feed('<hr>')
>>> ec.close()
>>> ec.get_events()
[('starttag', 'hr', [])]
>>> ec = EventCollector()
>>> ec.feed('<hr/>')
>>> ec.close()
>>> ec.get_events()
[('startendtag', 'hr', [])]
```

Reply:
So I see that `<hr />` fails in a different way:

```html
<p>foo
    <hr />
</p>
<p>bar</p>
```

Sigh. Apparently it never occurred to me we would need separate tests specifically for `hr` tags.